src/cryptonote_core/blockchain.cpp

```diff
@@ -3881,6 +3881,15 @@ bool Blockchain::check_tx_inputs(transaction& tx,
     sig_index++;
   }
 
+  if (rct::is_rct_fcmp(tx.rct_signatures.type))
+  {
+    // Set max used block height to the youngest block which contains outputs in the referenced FCMP tree
+    static_assert(CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE >= 1);
+    assert(nullptr != pmax_used_block_height); // set earlier
+    *pmax_used_block_height = tx.rct_signatures.p.reference_block
+      - std::min<std::uint64_t>(tx.rct_signatures.p.reference_block, CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE - 1);
```

Comment:
Note for self:

I use a similar pattern in db_lmdb.cpp [here](https://github.com/seraphis-migration/monero/blob/28fdf9829835da7b8c5daab2b800109922774140/src/blockchain_db/lmdb/db_lmdb.cpp#L7402)

I think it would be good to use a helper function for this logic just so it's consolidated somewhere / makes it easier to manage if there is some change down the road

I can do it in a separate PR